pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -276,6 +273,14 @@ func (esc *endpointSliceController) updateEndpointCacheForSlice(hostName host.Na
 				}
 
 				istioEndpoint := builder.buildIstioEndpoint(a, portNum, portName, discoverabilityPolicy, healthStatus)
+				if features.EnableDualStack && (pod != nil) && len(pod.Status.PodIPs) > 1 {
```

Comment:
This seems wrong. This will make all services dual stack, even if they are explicitly single stack

Reply:
Makes sense to disable that logic for DS Svcs but I have one doubt on how to perform the check given the current service.go APIs.
I can't use GetExtraAddressesForProxy (as the update is not proxy-specific), shall I add a new hasExtraAddresses() function in our service model to verify if **any** of the clusters have an extra address for the service?

Reply:
This code is single cluster, so I am  not sure I understand why we need this info. Is it because we do not have access to the `corev1.Service` in this part of the code?

Reply:
Currently that part of the code obtains the model.Service, but I think I can infer the k8s svc from the slice, let me try and test it :+1:

Reply:
I think the last commit should do the trick, unit tests for istio endpoints with multiple addresses are also passing. Tomorrow I will deploy it and verify that it works in a live env.

Reply:
It is working fine after the last commit.
I have added two test cases for single-stack svc on DS deployments to have some coverage of this.

```
0.244.0.28:18080,fd00:10:244::1c:18080          HEALTHY                             outbound|80||echo-dual.echo-ds-3-3070.svc.cluster.local
10.244.0.22:18080,fd00:10:244::16:18080          HEALTHY                             outbound|80||echo-v4-naked.echo-ds-3-3070.svc.cluster.local
10.244.0.25:18080,fd00:10:244::19:18080          HEALTHY                             outbound|80||echo-v4.echo-ds-3-3070.svc.cluster.local
10.244.0.27:18080                                HEALTHY                             outbound|80||echo-v4onlysvc.echo-ds-3-3070.svc.cluster.local
10.244.0.21:18080,fd00:10:244::15:18080          HEALTHY                             outbound|80||echo-v6-naked.echo-ds-3-3070.svc.cluster.local
10.244.0.23:18080,fd00:10:244::17:18080          HEALTHY                             outbound|80||echo-v6.echo-ds-3-3070.svc.cluster.local
fd00:10:244::18:18080                            HEALTHY                             outbound|80||echo-v6onlysvc.echo-ds-3-3070.svc.cluster.local
```